src/Chapter1.hs

```diff
@@ -653,8 +680,11 @@ You need to use recursion in this task. Feel free to return to it later, if you
 aren't ready for this boss yet!
 -}
 
-firstDigit n = error "firstDigit: Not implemented!"
-
+firstDigit :: Int -> Int
+firstDigit n
+    | mod (abs n) 10 == abs n = abs n
+    | otherwise = firstDigit (div (abs n) 10)
```

Comment:
You can see that `abs n` is reused several times in this function. This usually means that you can move it into the `where` case and reuse everywhere 👍🏼 
But most efficiently would be to use `abs n` only once, and in the recursive functions not to use it at all.